### PR TITLE
Convert blowfish tests to new framework

### PR DESCRIPTION
**Pull request by jon-oracle**
##### Checklist
- [x] tests are added or updated

Converted the Blowfish tests.

### CODE REVIEW

File: test/bftest.c [hunk 1]
```diff
@@ -229,20 +232,25 @@ static unsigned char key_out[KEY_TEST_NUM][8] = {
...
         ret = print_test_data();
     else
-        ret = test();
+        ret = run_tests(argv[0]);
 
-    EXIT(ret);
+    return ret;
 }
 
+
```
Comment (mattcaswell) [inline test/bftest.c]: Remove spurious newline

File: test/bftest.c [hunk 2]
```diff
@@ -331,80 +339,37 @@ static int test(void)
...
-            for (i = 0; i < 2; i++)
-                printf("%08lX ", (unsigned long)data[i]);
-            printf("\n");
-            printf("expected:");
-            for (i = 0; i < 2; i++)
-                printf("%08lX ", (unsigned long)bf_cipher[n][i]);
-            err = 1;
-            printf("\n");
-        }
+        if (!TEST_mem_eq(&(bf_cipher[n][0]), 8, &(data[0]), 8))
```
Comment (mattcaswell) [inline test/bftest.c]: BF_BLOCK instead of 8? Here and elsewhere.

File: test/bftest.c [hunk 3]
```diff
@@ -331,80 +339,37 @@ static int test(void)
...
-            err = 1;
-        }
+        if (!TEST_mem_eq(&(bf_plain[n][0]), 8, &(data[0]), 8))
+            ret = 0;
     }
 
-    printf("testing blowfish in ecb mode\n");
-
+    /* ECB Mode */
     for (n = 0; n < NUM_TESTS; n++) {
```
Comment (mattcaswell) [inline test/bftest.c]: Should this loop be done with ADD_ALL_TESTS instead?

Comment (jon-oracle) [inline test/bftest.c]: Tests have now been split up to allow this

**Comment by levitte** [comment]
@mattcaswell, reconfirm?

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**